### PR TITLE
Bug fix volgrp

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
Modify for volume group was failing because expiration timestamp was getting passed due to volume group snapshot.
updated on gopowerstore side as well.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance tests

### CODE REVIEW

File: powerstore/resource_volume_group_snapshot.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func (r *resourceVGSnapshot) Update(ctx context.Context, req resource.UpdateRequ
 		return
 	}
 
-	if plan.VolumeGroupID.ValueString() != state.VolumeGroupID.ValueString() || plan.VolumeGroupName.ValueString() != state.VolumeGroupName.ValueString() {
+	if (plan.VolumeGroupID.ValueString() != "" && plan.VolumeGroupID.ValueString() != state.VolumeGroupID.ValueString()) || plan.VolumeGroupName.ValueString() != state.VolumeGroupName.ValueString() {
```
Comment (shenda1) [inline powerstore/resource_volume_group_snapshot.go]: What's the behavior when volume group name changes?

Comment (Sakshi-dell) [reply powerstore/resource_volume_group_snapshot.go]: It gives error stating
Volume group Name or Volume group ID cannot be updated

**Review by prajwalpatil25 — DISMISSED** [review]

File: powerstore/resource_volume_group_snapshot.go (hunk 1 above)
Comment (shenda1) [reply powerstore/resource_volume_group_snapshot.go]: Basically, we are restricting the update operation when the volume group name is changed, even though its valid operation

Comment (Sakshi-dell) [reply powerstore/resource_volume_group_snapshot.go]: No we are not restricting a valid operation. In volume group snapshot, volume group name and ID cannot be updated. 
In order to restrict this we have added a check that gives error when a user tries to update either of them.

**Review by shenda1 — DISMISSED** [review]

**Comment by Sakshi-dell** [comment]
image

**Review by shenda1 — APPROVED** [review]

**Review by prajwalpatil25 — APPROVED** [review]

**Pull request merged**